src/safeposix/syscalls/fs_calls.rs

```diff
@@ -4548,6 +4590,8 @@ impl Cage {
             // dropping the original reference lets us modify the value without deadlocking
             // the dashmap.
             drop(sementry);
+            // Acquire the semaphore. This operation will block the calling process until the 
```

Comment:
I'm not sure why this is being changed? I feel like I updated this, is your branch up to date?